kong/clustering/services/sync/rpc.lua

```diff
@@ -176,7 +177,23 @@ local function is_rpc_ready()
 end
 
 
-local function do_sync()
+local do_sync
+
+
+local function error_handler(err, is_full_sync)
```

Comment:
I think that this `error_handler()` is a premature optimization, we could have a simple function like:
```
try_sync(is_full_sync)
  if is_full_sync then
    yield()
    return do_sync()
  end
end
```
And keep the oringinal error handling code.

Reply:
This causes an unconditional yielding.

Reply:
may be
```
        if not res then
  if is_full_sync then
    return try_sync()
  end
          return nil, err
        end
```

Reply:
for sync I don't think that we should hide this operation in an error handler function, we should show it clearly.

Reply:
I will show you why this is designed like this in EE's PR